src/Control/Effect/Internal/Error.hs

```diff
@@ -234,6 +234,49 @@ errorToErrorIO main = do
     else
       throwIO exc
 
+data ErrorToErrorIOAsExcH
+
+instance (Eff ErrorIO m, Exception e) => Handler ErrorToErrorIOAsExcH (Throw e) m where
+  effHandler (Throw e) = throwIO e
+
+instance (Eff ErrorIO m, Exception e) => Handler ErrorToErrorIOAsExcH (Catch e) m where
+  effHandler (Catch m f) =  m `catchIO` f
+
+newtype ErrorToErrorIOAsExcC e m a = ErrorToErrorIOAsExcC  {
+    unErrorToErrorIOAsExcC ::
+      InterpretC ErrorToErrorIOAsExcH (Catch e)
+        (InterpretC ErrorToErrorIOAsExcH (Throw e) m) a
+  } deriving ( Functor, Applicative, Monad
+             , Alternative, MonadPlus
+             , MonadFix, MonadFail, MonadIO
+             , MonadThrow, MonadCatch, MonadMask
+             , MonadBase b, MonadBaseControl b
+             )
+    deriving (MonadTrans, MonadTransControl)
+    via CompositionBaseT
+     '[ InterpretC ErrorToErrorIOAsExcH  (Catch e)
+      , InterpretC ErrorToErrorIOAsExcH (Throw e)
+      ]
+
+deriving instance (Eff ErrorIO m, Exception e) => Carrier (ErrorToErrorIOAsExcC e m)
+
+-- | Runs connected 'Throw' and 'Catch' effects -- i.e. 'Error' --
+-- by transforming them into 'ErrorIO'
+--
+-- @'Derivs' ('ErrorToIOAsExcC' e m) = 'Catch' e ': 'Throw' e ': 'Derivs' m@
+--
+-- @'Control.Effect.Primitive.Prims' ('ErrorToIOAsExcC' e m) = 'Control.Effect.Optional.Optional' ((->) 'Control.Exception.SomeException') ': 'Control.Effect.Primitive.Prims' m@
```

Comment:
Sorry, should've been more clear: only `errorToIOAsExc` has `Optional SomeException` as a primitive effect; `errorToErrorIOAsExc` doesn't.

```haskell
-- @'Derivs' ('ErrorToErrorIOAsExcC' e m) = 'Catch' e ': 'Throw' e ': 'Derivs' m@
--
-- @'Control.Effect.Primitive.Prims' ('ErrorToErrorIOAsExcC' e m) = 'Control.Effect.Primitive.Prims' m@
````

Reply:
Ah right my bad